File: desktopclient/src/main/java/com/goexp/galgame/gui/view/common/jump/SearchController.java
package com.goexp.galgame.gui.view.common.jump;

import com.goexp.galgame.common.website.GGBasesURL;
import com.goexp.galgame.common.website.WikiURL;
import com.goexp.galgame.common.website._2DFURL;
import com.goexp.galgame.gui.view.HGameApp;
import javafx.fxml.FXML;
import javafx.scene.control.MenuItem;


public class SearchController {

    private String keyword = "";

//    public MenuButton searchLink;

//    @FXML
//    private MenuItem linkDlsite;

    @FXML
    private MenuItem linkGGBases;

    @FXML
    private MenuItem linkWiki;

    @FXML
    private MenuItem link2DF;

//    @FXML
//    private MenuItem linkBangumi;

    @FXML
    private void initialize() {

//        linkDlsite.setOnAction((e) -> {
//            HGameApp.app.getHostServices().showDocument(DlSiteURL.fromTitle(keyword));
//        });
        linkGGBases.setOnAction((e) -> {
            HGameApp.app.getHostServices().showDocument(GGBasesURL.fromTitle(keyword));
        });
        linkWiki.setOnAction((e) -> {
            HGameApp.app.getHostServices().showDocument(WikiURL.fromTitle(keyword));
        });
        link2DF.setOnAction((e) -> {
            HGameApp.app.getHostServices().showDocument(_2DFURL.fromTitle(keyword));
        });
//        linkBangumi.setOnAction((e) -> {
//            HGameApp.app.getHostServices().showDocument(BangumiURL.fromTitle(keyword));
//        });
    }

    public void load(String keyword) {
        this.keyword = keyword;
    }


}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/view/common/jump/SearchController.scala
package com.goexp.galgame.gui.view.common.jump

import com.goexp.galgame.common.website.{ErogameScapeURL, GGBasesURL, WikiURL, _2DFURL}
import com.goexp.galgame.gui.util.Websites
import com.goexp.ui.javafx.DefaultController
import javafx.beans.property.SimpleStringProperty
import javafx.fxml.FXML
import javafx.scene.control.MenuItem

class SearchController extends DefaultController {
  final lazy val keyword = new SimpleStringProperty()

  @FXML private var linkGGBases: MenuItem = _
  @FXML private var linkWiki: MenuItem = _
  @FXML private var linkEgs: MenuItem = _
  @FXML private var link2DF: MenuItem = _

  override protected def eventBinding() = {
    linkGGBases.setOnAction(_ => Websites.open(GGBasesURL.fromTitle(keyword.get())))
    linkWiki.setOnAction(_ => Websites.open(WikiURL.fromTitle(keyword.get())))
    linkEgs.setOnAction(_ => Websites.open(ErogameScapeURL.fromTitle(keyword.get())))
    link2DF.setOnAction(_ => Websites.open(_2DFURL.fromTitle(keyword.get())))
  }

}